pyntrinio/pyntrinio.py

```diff
@@ -189,15 +189,65 @@ def gather_stock_time_series(api_key, ticker, start_date, end_date, output_forma
   Example
   -----------
   >>> gather_stock_time_series(api_key, 'AAPL')
-  {'ticker':['AAPL'], 'adj_close':[300], 'date':[datetime.date(2020, 1, 2)]}
   """
+    # ensure the type of the ticker is a string
+    if type(ticker) != str:
+        print("Invalid Input: ticker has to be a string, e.g. 'APPL'")
+        return
+    
+    try:
+        # change dates to datetime objects
+        if start_date is not None:
+            start_date = datetime.strptime(start_date, '%Y-%m-%d').date()
+        if end_date is not None:
+            end_date = datetime.strptime(end_date, '%Y-%m-%d').date()
+    except:
+        print("Invalid Date format - please input the date as a string with format %Y-%m-%d")
+        return
+    
+    if start_date >= end_date:
+        print("Invalid Input: end_date is earlier than start_date")
+        return
   
-  if output_format=='dict':
-    results={}
-  else:
-    results=pd.DataFrame(results)
-  
-  return results
+    try:
+        # initialize API key
+        intrinio_sdk.ApiClient().configuration.api_key['api_key'] = api_key
+    
+    except:
+        print("Incorrect API Key - please input a valid API key as a string")
+        return
```

Comment:
Even if the `api_key` is not valid, it won't give errors at this stage.  My code for the reference:  
```
  # initialize API key
  intrinio_sdk.ApiClient().configuration.api_key['api_key'] = api_key
  
  # initialize security API
  security_api = intrinio_sdk.SecurityApi()
  
  # test if the API Key works
  try:
    security_api.get_security_stock_prices(ticker[0], start_date=buy_date, end_date=sell_date)
  except:
    print("Incorrect API Key - please input a valid API key as a string")
    return
```

Reply:
done